pythonwhat/test_function_definition.py

```diff
@@ -272,15 +299,52 @@ def test_args(rep, arg_names, arg_defaults, args_student, args_solution,
                     EqualTest(arg_name_solution, arg_name_student, Feedback(c_arg_names_msg, fun_def)))
                 if rep.failed_test:
                     return
+
             if arg_defaults:
-                c_arg_defaults_msg = arg_defaults_msg or \
-                    ("In your definition of %s, the %s argument should have `%s` as default, instead got `%s`." %
-                        (name, get_ord(i+1), arg_default_solution, arg_default_student))
-                rep.do_test(
-                    EqualTest(arg_default_solution, arg_default_student, Feedback(c_arg_defaults_msg, fun_def)))
+
+                if arg_defaults_msg is None:
+                    if arg_default_solution is None:
+                        c_arg_defaults_msg = "In your definition of %s, the argument `%s` should have no default." % (name, arg_name_student)
+                    else :
+                        c_arg_defaults_msg = "In your definition of %s, the argument `%s` does not have the correct default." % (name, arg_name_student)
+                else :
+                    c_arg_defaults_msg = arg_defaults_msg
+
+                if arg_default_solution is None:
```

Comment:
I was thinking here it would be more clear if you worked with `elif` and longer conditions.
```
if arg_default_solution is None and arg_default_student is not None:
...
elif arg_default_solution is not None and arg_default_student is None:
...
elif arg_default_solution is not None and arg_default_student is None:
...
```